pandas/tests/series/test_datetime_values.py

```diff
@@ -282,6 +282,27 @@ def test_dt_round_tz_ambiguous(self, method):
         with pytest.raises(pytz.AmbiguousTimeError):
             getattr(df1.date.dt, method)('H', ambiguous='raise')
 
+    @pytest.mark.parametrize('method, ts_str, freq', [
+        ['ceil', '2018-03-11 01:59:00-0600', '5min'],
+        ['round', '2018-03-11 01:59:00-0600', '5min'],
+        ['floor', '2018-03-11 03:01:00-0500', '2H']])
+    def test_dt_round_tz_nonexistent(self, method, ts_str, freq):
+        # GH 23324 round near "spring forward" DST
+        s = Series([pd.Timestamp(ts_str, tz='America/Chicago')])
+        result = getattr(s.dt, method)(freq, nonexistent='shift')
+        expected = Series(
+            [pd.Timestamp('2018-03-11 03:00:00', tz='America/Chicago')]
+        )
+        tm.assert_series_equal(result, expected)
+
+        result = getattr(s.dt, method)(freq, nonexistent='NaT')
+        expected = Series([pd.NaT]).dt.tz_localize(result.dt.tz)
+        tm.assert_series_equal(result, expected)
+
+        with pytest.raises(pytz.NonExistentTimeError,
+                           message='2018-03-11 02:00:00'):
+            getattr(s.dt, method)(freq, nonexistent='raise')
```

Comment:
* I'm almost inclined to have parameterization on `nonexistent`, but I'll leave that up to you.
* More general question, to what extent are we using `pytest.raises` vs `tm.assert_raises_regex` ?

cc @jreback

Reply:
I'm inclined to keep the format of this test. IMO parameterization over nonexistent would obfuscate the test too much.

Reply:
No problem.  Just a thought. 👍 

More general question, to what extent are we using `pytest.raises` vs `tm.assert_raises_regex` ?

cc @jreback

Reply:
ok with ``pytest.raises`` to the extent we are just checking the type of the error. so ok here

Reply:
@jreback : `pytest.raises` is also checking the error message here...